addons/gamification/models/challenge.py

```diff
@@ -543,42 +538,49 @@ def report_progress(self, users=(), subset_goals=False):
 
         MailTemplates = self.env['mail.template']
         if challenge.visibility_mode == 'ranking':
+            template_id = self.env.ref('gamification.simple_report_template_ranking', raise_if_not_found=False)
             lines_boards = challenge._get_serialized_challenge_lines(restrict_goals=subset_goals)
+            ctx = {
+                'challenge_lines': lines_boards,
+                'default_use_template': bool(template_id),
+                'default_template_id': template_id,
+                'custom_layout': "gamification.challenge_mail_template",
+            }
+            body_html = MailTemplates.with_context(ctx).render_template(template_id.body_html, 'gamification.challenge', challenge.id)
```

Comment:
will crash if the template is not found (so `raise_if_not_found` should probably stay at `True` actually